src/main/java/org/schemaspy/cli/DefaultProviderFactory.java

```diff
@@ -57,32 +65,35 @@ private static boolean exists(String propertiesFilename) {
      * If a properties file exists it will return CombinedDefaultProvider search order Properties then Env
      * If this file does not exist the method returns CombinedDefaultProvider with only EnvDefaultProvider
      *
-     * @param propertiesFilename
      * @return IDefaultProvider
      */
-    public IDefaultProvider create(String propertiesFilename) {
-        if (propertiesFilename != null) {
-            if (exists(propertiesFilename)) {
-				LOGGER.info("Found configuration file: {}", propertiesFilename);
+    public IDefaultProvider defaultProvider() {
+        if (propertiesFileName.isPresent()) {
+            String fileName = propertiesFileName.get();
+            if (exists(fileName)) {
+				LOGGER.info("Found configuration file: {}", fileName);
                 return new CombinedDefaultProvider(
-                        new PropertyFileDefaultProvider(propertiesFilename),
+                        new PropertyFileDefaultProvider(fileName),
                         new EnvDefaultProvider()
                 );
             } else {
-				LOGGER.error("Could not find config file: {}", propertiesFilename);
-                System.exit(0);
-                return null;
+				LOGGER.error("Could not find config file: {}", fileName);
+                throw new InvalidConfigurationException("File not found", new FileNotFoundException(fileName),"-configFile", fileName);
             }
         }
 
-        if (exists(DEFAULT_PROPERTIES_FILE_NAME)) {
-            LOGGER.info("Found configuration file: {}",DEFAULT_PROPERTIES_FILE_NAME);
+        if (exists(defaultPropertiesFileName)) {
+            LOGGER.info("Found default configuration file: {}",defaultPropertiesFileName);
             return new CombinedDefaultProvider(
-                    new PropertyFileDefaultProvider(DEFAULT_PROPERTIES_FILE_NAME),
+                    new PropertyFileDefaultProvider(defaultPropertiesFileName),
                     new EnvDefaultProvider()
             );
         }
         return new CombinedDefaultProvider(new EnvDefaultProvider());
     }
 
+    private static boolean exists(String fileName) {
+        return new File(fileName).exists();
```

Comment:
Ignoring rule in .lift.toml since we work with user input and we cant restrict were they put their files.

Another argument is that the input is not supplied from an external threat actor.